lib/symlink.go

```diff
@@ -1,27 +1,31 @@
 package lib
 
 import (
+	"errors"
+	"fmt"
 	"io"
 	"os"
+	"path/filepath"
 	"runtime"
+	"strings"
 )
 
 // CreateSymlink : create symlink or copy file to bin directory if windows
-func CreateSymlink(cwd string, dir string) {
+func CreateSymlink(cwd string, dir string) error {
 	// If we are on windows the symlink is not working correctly.
 	// Copy the desired terraform binary to the path environment.
 	if runtime.GOOS == "windows" {
 		r, err := os.Open(cwd)
 		if err != nil {
-			logger.Fatalf("Unable to open source binary: %s", cwd)
-			os.Exit(1)
+			logger.Debugf("Unable to open source binary: %s", cwd)
+			return errors.New(fmt.Sprintf("Unable to open source binary: %s", cwd))
```

Comment:
Would it make sense to log and/or return `err` also?

Reply:
Also why `errors.New(fmt.Sprintf(…))` rather than just `fmt.Errorf()`?
Ref: https://pkg.go.dev/errors#New